src/packageurl/contrib/purl2url.py

```diff
@@ -225,6 +225,23 @@ def build_pypi_repo_url(purl):
         return f"https://pypi.org/project/{name}/"
 
 
+@repo_router.route("pkg:composer/.*")
+def build_composer_repo_url(purl):
+    """
+    Return a composer repo URL from the `purl` string.
+    """
+    purl_data = PackageURL.from_string(purl)
+
+    name = (purl_data.name or "").replace("_", "-")
```

Comment:
I'm that sure that an empty name is ever possible, see https://github.com/camillem/packageurl-python/blob/e0d180d1ce23f289230a970c2d1e560456300bb8/src/packageurl/__init__.py#L542

Reply:
Indeed. I just directly copy/pasted it from https://github.com/camillem/packageurl-python/blob/e0d180d1ce23f289230a970c2d1e560456300bb8/src/packageurl/contrib/purl2url.py#L219
I should also check if the `replace("_", "-")` makes sense in this context.

Reply:
I simplified this part, as the `replace("_", "-")`  seemed to be linked to an old pypi.org oddity.